crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -98,12 +98,17 @@ fn get_operations(dir: impl AsRef<Path>, cloud_name: &str) -> Result<Operations,
     let mut operations = Operations::default();
 
     let path = dir.as_ref().join(&cloud_name);
-    let dir_entries = fs::read_dir(&path)?
-        .map(|entry| entry.map(|e| e.path()))
-        .collect::<Result<Vec<PathBuf>, _>>()?
-        .into_iter()
-        .filter(|path| path.is_file())
-        .collect::<Vec<PathBuf>>();
+    let dir_entries = match fs::read_dir(&path) {
+        Ok(entry) => entry
+            .map(|entry| entry.map(|e| e.path()))
+            .collect::<Result<Vec<PathBuf>, _>>()?
+            .into_iter()
+            .filter(|path| path.is_file())
+            .collect::<Vec<PathBuf>>(),
+        Err(_err) => {
+            return Err(OperationsError::InvalidOperationName(PathBuf::from(&path)));
+        }
+    };
```

Comment:
It's up to the preference, but you can also write like this without using `match`.

```suggestion
    let dir_entries = fs::read_dir(&path)
        .map_err(|_| OperationsError::InvalidOperationName(PathBuf::from(&path)))?
        .map(|entry| entry.map(|e| e.path()))
        .collect::<Result<Vec<PathBuf>, _>>()?
        .into_iter()
        .filter(|path| path.is_file())
        .collect::<Vec<PathBuf>>();
```